src/ptr_operator.rs

```diff
@@ -2,33 +2,41 @@ use zkevm_opcode_defs::MAX_OFFSET_FOR_ADD_SUB;
 
 use crate::{
     address_operands::{address_operands_read, address_operands_store},
+    eravm_error::EraVmError,
     state::VMState,
     value::{FatPointer, TaggedValue},
     Opcode,
 };
 
-pub fn ptr_operands_read(vm: &mut VMState, opcode: &Opcode) -> (FatPointer, u32, TaggedValue) {
-    let (src0, src1) = address_operands_read(vm, opcode);
+pub fn ptr_operands_read(
+    vm: &mut VMState,
+    opcode: &Opcode,
+) -> Result<(FatPointer, u32, TaggedValue), EraVmError> {
+    let (src0, src1) = address_operands_read(vm, opcode)?;
 
     if !src0.is_pointer || src1.is_pointer {
-        panic!("Invalid operands for {:?}", opcode.variant);
+        return Err(EraVmError::OperandError(
+            "Invalid operands for ptr_shrink".to_string(),
```

Comment:
Maybe `OperandError` could contain the opcode variant instead.

Reply:
You are right, adding it.